### DISCUSSION TITLE
packaging-a-go-app-for-nixos/

### DISCUSSION THREAD

**Discussion by giscus** [ironicbadger/blog.ktz.me#228] [Announcements]
# packaging-a-go-app-for-nixos/

The imposter syndrome around almost anything to do with NixOS is as strong as the pull to use it in the first place.

https://blog.ktz.me/packaging-a-go-app-for-nixos/

**Comment by oscarcarlsson** [comment]
You can also define an overlay for your custom package, in case you can't publish it to nixpkgs (doing it is awesome, tho!). It's pretty easy, I wrote a note on how to do it for future me here: https://www.monotux.tech/posts/2023/05/nixos-overlay/

Then the package is available locally as if it had been in nixpkgs, so you can install it, use it in a custom systemd service and so forth.

**Reply by ironicbadger** [reply]
that's awesome!

**Reply by KimVatnedal** [reply]
Brilliant :)

Worked perfectly. Now I'm finally able to distinguish between my Nix servers as well... ;)
